homeassistant/components/enphase_envoy/sensor.py

```diff
@@ -446,6 +564,40 @@ def native_value(self) -> int | None:
         return self.entity_description.value_fn(system_consumption)
 
 
+class EnvoyProductionPhaseEntity(EnvoySystemSensorEntity):
+    """Envoy phase production entity."""
+
+    entity_description: EnvoyProductionPhaseSensorEntityDescription
+
+    @property
+    def native_value(self) -> int | None:
+        """Return the state of the sensor."""
+        if self.data.system_production_phases is not None:
```

Comment:
Can you maybe elaborate a bit more about why we do this? Is this a way to remove an assert (which we have too many of, I agree)?

Or does this mean we only show unknown at devices which don't support this? When can this be None?

(this question also applies for the Consumption one)

(if this is intended, I think inverting it would be better)

```suggestion
        if self.data.system_production_phases is None:
            return None
        ...
```

Reply:
Yes they can be none and will be none for Envoy without CT or non-metered versions. I'll apply your reverted proposal and do the same for the one below to get rid of the assert
```py
            if system_production is None:
                return None
            return self.entity_description.value_fn(system_production)
```

Reply:
But if someone has no CT, shouldn't we omit these entities from being added in the first place? This way they aren't fed up with X amount of entities that are Unknown

Reply:
Yes that is the intent. If none is returned no entity is created.

Reply:
refactored it